### PR TITLE
Smoldering Heart and Stormbringer

### PR DESCRIPTION
**Pull request by Zekulio**
Smoldering Heart pulled from elemental and edited to work for enhance.
Stormbringer copied from frost mage brain freeze and edited.
Corrected spell id for Stormbringer_Buff.

### CODE REVIEW

File: src/Parser/Shaman/Enhancement/Modules/Items/SmolderingHeart.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          this.maelstromSpent += 30;
+      } else if (event.ability.guid === SPELLS.CRASH_LIGHTNING.id || event.ability.guid === SPELLS.EARTHEN_SPIKE_TALENT.id || event.ability.guid === SPELLS.FROSTBRAND.id || event.ability.guid === SPELLS.SUNDERING_TALENT.id) {
+          this.maelstromSpent += 20;
+      } else if (event.ability.guid === SPELLS.WINDSTRIKE.id) {
+          this.maelstromSpent += 8; //8 due to 80% cost reduction in Ascendance
+      }
+  }
+
+  on_byPlayer_applybuff(event) {
+    if (event.ability.guid === SPELLS.ASCENDANCE_TALENT_ENHANCEMENT.id) {
```
Comment (CollCrom) [inline src/Parser/Shaman/Enhancement/Modules/Items/SmolderingHeart.js]: I am not 100% on this but cant you look for ascendance applyBuffs that didnt follow a ascendance cast?

File: src/Parser/Shaman/Enhancement/Modules/Items/SmolderingHeart.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        if (!this.ascendanceCastEventIds.has(event.eventUniqueId + 1)) {
+            // On manual Ascendance cast, the applybuff event occurs one event before the cast event, so checking if
+            // a cast event exists one place after the applybuff event is a robust way of determining whether an Ascendance
+            // buff was manually casted or is a result of a Smoldering Heart proc.
+            // Magic number: 10000ms is the duration of the Smoldering Heart Asc proc.
+        }
+        this.smolderingProcs += 1;
+        this.smolderingDuration += 10000;
+    }
```
Comment (CollCrom) [inline src/Parser/Shaman/Enhancement/Modules/Items/SmolderingHeart.js]: Why not use the const object at the beginning?

**Comment by CollCrom** [comment]
Also there are some local travis errors

**Comment by MartijnHols** [comment]
@Zekulio Thanks for your contribution. Unfortunately as @CollCrom pointed out we still have some questions and suggestions before we would be able to merge this into our project. The most important thing is to ensure it's accurate and not misleading, and unfortunately we do not have sufficient class knowledge to answer these questions ourselves.

Because this PR has had no activity for a month I will close it now. If you wish to take a stab at it you can still post a comment in this PR or reopen it, or create a new PR when you're ready.

Thank you for your time.

**Pull request closed without merging**